src/com/google/enterprise/adaptor/DocIdPusher.java

```diff
@@ -184,6 +211,46 @@ public GroupPrincipal pushGroupDefinitions(
       throws InterruptedException;
 
   /**
+   * Blocking call to push group definitions to GSA ends in success or
+   * when provided error handler gives up.  Can take significant time
+   * if errors arise.
+   *
+   * <p>A group definition consists of a group being defined
+   * and members, which is a list of users and groups.
+   *
+   * <p>If you plan on using the return code, then the provided map should have
+   * a predictable iteration order, like {@link java.util.TreeMap}.
+   *
+   * <p>An incremental push augments or updates any existing group definitions
+   * for the groupSource on the GSA. Otherwise, the supplied group definitions
+   * will wholly replace all existing definitions for the groupSource on the
+   * GSA.
+   *
+   * <p>A groupSource identifies the source of the group definitions.
+   * If provided, it must be a string of the form
+   * <code>[a-zA-Z_][a-zA-Z0-9_-]*</code> with at most 250 characters.
+   * If groupSource is {@code null}, then the {@code feed.name} configuration
+   * property is used.
+   *
+   * <p>If handler is {@code null}, then a default error handler is used.
+   *
+   * @param defs map of group definitions
+   * @param caseSensitive when comparing Principals
+   * @param feedType if INCREMENTAL, an incremental update is done; if FULL, a
+   *        full replacement is done.
+   * @param groupSource
```

Comment:
This will allow the SharePoint connector to push a separate full group feed for each SharePoint site collection, since each has its own distinct groups.